__tests__/BankAccountsApi.spec.ts

```diff
@@ -68,38 +68,64 @@ describe("BankAccountsApi", () => {
     beforeAll(async () => {
       const bankApi = new BankAccountsApi(CONFIG_FOR_INTEGRATION);
 
-      // ensure there are at least 3 cards present, to test pagination
-      const bank2: BankAccountWritable = Object.assign({}, dummyAccount, {
-        signatory: "Juanita Lupo",
-      });
-      const bank3: BankAccountWritable = Object.assign({}, dummyAccount, {
-        signatory: "Jeanette Leloup",
-      });
-      createdBankAccounts.push((await bankApi.create(dummyAccount)).id);
-      createdBankAccounts.push((await bankApi.create(bank2)).id);
-      createdBankAccounts.push((await bankApi.create(bank3)).id);
+      // Create enough bank accounts to ensure pagination works
+      const bankAccountsToCreate = [
+        dummyAccount,
+        Object.assign({}, dummyAccount, { signatory: "Juanita Lupo" }),
+        Object.assign({}, dummyAccount, { signatory: "Jeanette Leloup" }),
+        Object.assign({}, dummyAccount, { signatory: "John Smith" }),
+        Object.assign({}, dummyAccount, { signatory: "Jane Doe" }),
+        Object.assign({}, dummyAccount, { signatory: "Bob Johnson" }),
+      ];
+
+      // Create all bank accounts
+      try {
+        const creationPromises = bankAccountsToCreate.map(
+          async (bankAccount) => {
+            try {
+              const created = await bankApi.create(bankAccount);
+              return created.id;
+            } catch (error) {
+              console.log(`Failed to create bank account: ${error}`);
+              return null;
+            }
+          }
+        );
+
+        const createdIds = await Promise.all(creationPromises);
+        // Filter out any failed creations
+        createdBankAccounts.push(
+          ...createdIds.filter((id): id is string => id !== null)
+        );
+      } catch (error) {
+        console.log(`Error during bank account creation: ${error}`);
+      }
+
+      // Wait a moment for API processing
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // Get pagination data with a small limit to force pagination
+      const response = await bankApi.list(3);
+
+      // Verify we have pagination data
+      expect(response).toEqual(
+        expect.objectContaining({
+          data: expect.arrayContaining([expect.any(Object)]),
```

Comment:
Can we test against internal object structure in the array? Instead of matching against any object.